node/src/Producer.ts

```diff
@@ -603,14 +618,84 @@ export class Producer<ProducerAppData extends AppData = AppData>
 	}
 }
 
+export function producerTypeFromFbs(type: FbsRtpParameters.Type): ProducerType
+{
+	switch (type)
+	{
+		case FbsRtpParameters.Type.SIMPLE:
+			return 'simple';
+		case FbsRtpParameters.Type.SIMULCAST:
+			return 'simulcast';
+		case FbsRtpParameters.Type.SVC:
+			return 'svc';
+		default:
+			throw new TypeError(`invalid ProducerType: ${type}`);
```

Comment:
It's not an invalid "ProducerType" but an invalid "FbsRtpParameters.Type".

Reply:
As I commented [here](https://github.com/versatica/mediasoup/pull/1164/files#diff-ce1ef95b294b370ff3bc7484654d563c89a6faf7888bc49b819b2cd8443270d3R174) I believe we should have another type `ProducerType` which does not have the PIPE entry. Since a producer will never be of type PIPE.

Reply:
That's irrelevant here. I just meant that what it's invalid is `FbsRtpParameters.Type` (the `type` passed to the function) rather then `ProducerType` (which is the **result** of that function).

Reply:
Transport.Consume() receives a `producerId` and a `type` arguments, among others.

The `type` is the type of the Producer being consumed unless we want to create a **pipe** Consumer. Hence, IMHO rather than having such arguments we should have `producerId` and an optional `pipe` argument. Take the `type` from the Producer (giving it's id) and use it to choose which type of Consumer we need to create. If `pipe` is enabled then we ignore the Producer type and create a PipeConsumer.

This is not a blocker for this PR but I'll create a TODO in flatbuffers PR.

Reply:
Trying to clarify agsin: I just lilterally meant the log text. It said "invalid ProducerType: ${type}". However it was wrong. It should have been "invalid FbsRtpParameters.Type: ${type}" becuase that's what `type` is.

Reply:
Regarding pipe stuff: Could we please not enter into that thing now? For v4 I'd like to have a single `Consumer` class that can dynamically handle any number of streams and layers and where "pipe" is just an option to make the Consumer select a single incoming stream or deliver them all to the remote endpoint.